### ISSUE TITLE
Hats on HOTAS are mapped as d-pad

### ISSUE THREAD

**Issue by Changryy** [godotengine/godot#83322]
### Godot version

v4.2.beta.custom_build [a574c0296]

### System information

Windows 11 - Logitech G X52

### Issue description

Some game controllers have hats. A hat works like a d-pad but should NOT be mapped as a d-pad. This results in conflicting bindings as seen in #60862.
The offending code is in `Input::joy_hat` which is in `core\input\input.cpp`

### Steps to reproduce

Not relevant

### Minimal reproduction project

Not relevant

**Comment by AThousandShips** [comment]
In the future please do not edit the issue template, do not remove the sections

**Comment by Nintorch** [comment]
Hello! I feel like this isn't a bug, but rather a feature that doesn't exist in Godot yet, so personally I feel it would be better for it to be a feature proposal. May I ask what do you think? Should this issue be closed in favor of https://github.com/godotengine/godot-proposals/issues/9850 ?

**Comment by Changryy** [comment]
Yeah youre right. Its not as much a bug as it is a problem caused by the limitations of the current implementation. I however wouldnt want to close the issue *in favour of* the proposal because the proposal is just one of many ways to solve the issue and Im open to alternative solutions if anyone has other ideas for how to solve it
